src/rdysetup.c

```diff
@@ -376,8 +379,10 @@ static PetscErrorCode InitBoundaries(RDy rdy) {
   }
 
   // add an additional boundary for unassigned boundary edges if needed
-  PetscInt num_unassigned_edges, num_global_unassigned_edges;
-  PetscCall(ISGetLocalSize(unassigned_edges_is, &num_unassigned_edges));
+  PetscInt num_unassigned_edges = 0, num_global_unassigned_edges;
+  if (boundary_edge_present) {
+    PetscCall(ISGetLocalSize(unassigned_edges_is, &num_unassigned_edges));
+  }
   MPI_Allreduce(&num_unassigned_edges, &num_global_unassigned_edges, 1, MPI_INT, MPI_SUM, rdy->comm);
   if (num_global_unassigned_edges > 0) {
```

Comment:
@jeff-cohere I think we should use `if (num_unassigned_edges > 0)` instead of `if (num_global_unassigned_edges > 0)`. What do you think?

Reply:
In principle, it depends on whether `DMLabelSetStratumIS` has to be called the same way on all MPI ranks. But based on the fact that the stratum IS can be be `NULL` for some processes and not others (in other words, that an empty stratum IS is represented by "no stratum IS"), I think you are correct.

Reply:
It depends on whether we need the same label values to exist on all processes. Sometimes there is collective code that assumes that all processes have the same set of label values.